app/Services/PolygonService.php

```diff
@@ -45,6 +49,58 @@ class PolygonService
         'num_trees',
     ];
 
+    public function createProjectPolygon($entity, $currentGeojson)
+    {
+        if (GeometryHelper::isFeatureCollectionEmpty($currentGeojson)) {
+            return;
+        }
+
+        $needsVoronoi = GeometryHelper::isOneOrTwoPointFeatures($currentGeojson);
+        if ($needsVoronoi) {
+            $pointWithEstArea = GeometryHelper::addEstAreaToPointFeatures($currentGeojson);
+            $currentGeojson = App::make(PythonService::class)->voronoiTransformation(json_decode($pointWithEstArea));
+        }
+
+        $convexHullWkt = GeometryHelper::getConvexHull($currentGeojson);
+        if ($convexHullWkt) {
+            $polygonGeometry = new PolygonGeometry();
+            $polygonGeometry->geom = DB::raw("ST_GeomFromText('" . $convexHullWkt . "')");
+            $polygonGeometry->save();
+
+            ProjectPolygon::create([
+                'poly_uuid' => $polygonGeometry->uuid,
+                'entity_type' => get_class($entity),
+                'entity_id' => $entity->id,
+                'last_modified_by' => Auth::user() ? Auth::user()?->id : 'system',
+                'created_by' => Auth::user() ? Auth::user()?->id : 'system',
```

Comment:
is it possible that the endpoint is going to be called without authentication?
if the answer is no, then `Auth::user()?->id` shouldn't have an alternative.

Reply:
It is possible that the endpoint receives a non-authenticated request. That's why we add that condition.